src/execution/index/art/prefix_handle.cpp

```diff
@@ -0,0 +1,102 @@
+#include "duckdb/execution/index/art/prefix_handle.hpp"
+
+#include "duckdb/execution/index/art/art.hpp"
+#include "duckdb/execution/index/art/node.hpp"
+
+namespace duckdb {
+
+PrefixHandle::PrefixHandle(const ART &art, const Node node)
+    : segment_handle(make_uniq<SegmentHandle>(Node::GetAllocator(art, PREFIX).GetHandle(node))) {
+	data = segment_handle->GetPtr();
+	ptr = reinterpret_cast<Node *>(data + Count(art) + 1);
+	segment_handle->MarkModified();
+}
+
+PrefixHandle::PrefixHandle(unsafe_unique_ptr<FixedSizeAllocator> &allocator, const Node node, const uint8_t count)
+    : segment_handle(make_uniq<SegmentHandle>(allocator->GetHandle(node))) {
+	data = segment_handle->GetPtr();
+	ptr = reinterpret_cast<Node *>(data + count + 1);
+	segment_handle->MarkModified();
+}
+
+PrefixHandle::PrefixHandle(PrefixHandle &&other) noexcept
+    : segment_handle(std::move(other.segment_handle)), data(other.data), ptr(other.ptr) {
+	other.data = nullptr;
+	other.ptr = nullptr;
+}
+
+PrefixHandle &PrefixHandle::operator=(PrefixHandle &&other) noexcept {
+	if (this != &other) {
+		segment_handle = std::move(other.segment_handle);
+		data = other.data;
+		ptr = other.ptr;
+
+		other.data = nullptr;
+		other.ptr = nullptr;
+	}
+	return *this;
+}
+
+PrefixHandle PrefixHandle::NewDeprecated(unsafe_unique_ptr<FixedSizeAllocator> &allocator, Node &node) {
+	node = allocator->New();
+	node.SetMetadata(static_cast<uint8_t>(PREFIX));
+
+	PrefixHandle handle(allocator, node, DEPRECATED_COUNT);
+	handle.data[DEPRECATED_COUNT] = 0;
+	return handle;
+}
+
+void PrefixHandle::TransformToDeprecated(ART &art, Node &node, unsafe_unique_ptr<FixedSizeAllocator> &allocator) {
+	// Early-out, if we do not need any transformations.
+	if (!allocator) {
```

Comment:
Ah, I see now why we are passing a pointer to the allocator... That's very implicit. I wonder if we want to make it more explicit with, e.g., a `TransformState` or so? Which could hold multiple things, like the a flag, an optional pointer to the allocator, ...? What do you think? Then, we could still pass `FixedSizeAllocator &allocator` to all functions where we do not need to check for existence.